modAionImpl/src/org/aion/zero/impl/db/AionRepositoryImpl.java

```diff
@@ -332,8 +334,9 @@ public IDataWord getStorageValue(Address address, IDataWord key) {
         List<byte[]> rtn = new ArrayList<>();
         rwLock.readLock().lock();
         try {
-            Set<byte[]> keySet = txPoolDatabase.keys();
-            for (byte[] b : keySet) {
+            Iterator<byte[]> iterator = txPoolDatabase.keys();
+            while (iterator.hasNext()) {
+                byte[] b = iterator.next();
```

Comment:
The check is still needed because of the newly added snapshot logic. It's also somewhat required when using `Optional`. For example, the IDE code analyzer will give warnings when `get` is called without a preceding `isPresent` check.